### ISSUE TITLE
An infinite loop with ctre::range when matching input string

### ISSUE THREAD

**Issue by hanickadot** [hanickadot/compile-time-regular-expressions#222]
I thought I removed all cases 🤷🏻‍♀️

> ctre::range matches an empty string will cause infinite loop, e.g., for (auto&& m : ctre::range<"a|(ab)?">("aab"));

(description from another issue)

**Comment by heyingge-1560733** [comment]
Yeah. I found this causes some issue in my own project while using the existing regular expression pattern. Looks like you have been aware of it. Is it something to be fixed？

**Comment by hanickadot** [comment]
I need to investigate it more.

**Comment by heyingge-1560733** [comment]
Just a hint from what I studied, you can avoid that by doing the following -- if previous search is empty then add `PCRE_NOTEMPTY` option to the next search.
